components/list/test/list-item-expand-collapse-mixin.test.js

```diff
@@ -0,0 +1,144 @@
+import { defineCE, expect, fixture, oneEvent } from '@open-wc/testing';
+import { html, LitElement } from 'lit';
+import { ListItemCheckboxMixin } from '../list-item-checkbox-mixin.js';
+import { ListItemExpandCollapseMixin } from '../list-item-expand-collapse-mixin.js';
+import { ListItemMixin } from '../list-item-mixin.js';
+
+const tag = defineCE(
+	class extends ListItemMixin(ListItemExpandCollapseMixin(ListItemCheckboxMixin(LitElement))) {
+		render() {
+			return html`
+				${this._renderExpandCollapse()}
+				${this._renderExpandCollapseAction()}
+				${this._renderNestedLoadingSpinner()}
+			`;
+		}
+	}
+);
+
+describe('ListItemExpandCollapseMixin', () => {
+	it('Sets expandable to false when no key is given', async() => {
+		const element = await fixture(`<${tag} expandable></${tag}>`);
+		await element.updateComplete;
+		expect(element.expandable).to.be.false;
+	});
+
+	describe('Render expand/collapse action area when appropriate', () => {
+		const cases = [{
+			properties: { selectable: false, expandable: true, noPrimaryAction: false },
+			actionAreaAvailable: true
+		},
+		{
+			properties: { selectable: false, expandable: false, noPrimaryAction: false },
+			actionAreaAvailable: false
+		},
+		{
+			properties: { selectable: true, expandable: true, noPrimaryAction: false },
+			actionAreaAvailable: false
+		},
+		{
+			properties: { selectable: false, expandable: true, noPrimaryAction: true },
+			actionAreaAvailable: false
+		}];
+
+		for (const test of cases) {
+			it(`selectable: ${test.properties.selectable} expandable: ${test.properties.expandable} noPrimaryAction: ${test.properties.noPrimaryAction}`, async() => {
+				const element = await fixture(`<${tag} key="1234" expanded></${tag}>`);
+				for (const [key, value] of Object.entries(test.properties)) {
+					element[key] = value;
+				}
+				await element.updateComplete;
+				const actionArea = element.shadowRoot.querySelector('.d2l-list-expand-collapse-action');
+				if (test.actionAreaAvailable) {
+					expect(actionArea).to.exist;
+				} else {
+					expect(actionArea).to.not.exist;
+				}
+			});
+		}
+	});
+
+	describe('Render expand/collapse slot and button when appropriate', () => {
```

Comment:
Are these already effectively captured by the visual-diff tests? If so, I would be happy to have these pared down.